src/NUnitFramework/framework/Internal/Execution/EventListenerTextWriter.cs

```diff
@@ -264,25 +256,8 @@ public override void WriteLine(string value)
         /// </summary>
         public override void WriteLine(object value)
         {
-            if (value == null)
-            {
-                if (!TrySendLineToListener(string.Empty))
-                    _defaultWriter.WriteLine(value);
-            }
-            else
-            {
-                IFormattable f = value as IFormattable;
-                if (f != null)
-                {
-                    if (!TrySendLineToListener(f.ToString(null, FormatProvider)))
-                        _defaultWriter.WriteLine(value);
-                }
-                else
-                {
-                    if (!TrySendLineToListener(value.ToString()))
-                        _defaultWriter.WriteLine(value);
-                }
-            }
+            if (!TrySendLineToListener(FormatForListener(value)))
```

Comment:
I like how it cleans up the duplication.

Reply:
Only a suggestion. I could try to look at it as a switch statement and see how that works.